scipy/optimize/_differentialevolution.py

```diff
@@ -786,6 +775,12 @@ def solve(self):
             if warning_flag or intol:
                 break
 
+            if self.callback and self.callback(self.x, convergence=self.tol / self.convergence):
```

Comment:
@andyfaff thanks for you suggestion of a False test.

When I ran it I found that the division by self.convergence causes a zero division error if convergence is zero.

I moved this block down because I figured if I put it *after* convergence tests, we should never see the division by zero.

Reply:
Unfortunately the code needs to be moved back where it was. By moving it down the callback won't be called on the last iteration when convergence is reached.